### PR TITLE
net: lwm2m: Adds semantics for handling time

### PR DESCRIPTION
**Pull request by VeijoPesonen**
This far time values have been synonymous to integer values. Content formats like CBOR do use different representation. CBOR is introduced in LwM2M v1.1 so this is preparatory work that needs to be done first.

Signed-off-by: Veijo Pesonen <veijo.pesonen@nordicsemi.no>

### CODE REVIEW

File: subsys/net/lib/lwm2m/lwm2m_object.h [hunk 1]
```diff
@@ -500,6 +500,8 @@ struct lwm2m_writer {
 		       struct lwm2m_obj_path *path, int32_t value);
 	int (*put_s64)(struct lwm2m_output_context *out,
 		       struct lwm2m_obj_path *path, int64_t value);
+	int (*put_time)(struct lwm2m_output_context *out,
+		     struct lwm2m_obj_path *path, int64_t value);
```
Comment (rlubos) [inline subsys/net/lib/lwm2m/lwm2m_object.h:504]: nit: Please fix the alignment

File: subsys/net/lib/lwm2m/lwm2m_object.h [hunk 2]
```diff
@@ -686,6 +689,13 @@ static inline int engine_put_float(struct lwm2m_output_context *out,
 	return out->writer->put_float(out, path, value);
 }
 
+static inline int engine_put_time(struct lwm2m_output_context *out,
+				    struct lwm2m_obj_path *path,
+				    int64_t value)
```
Comment (rlubos) [inline subsys/net/lib/lwm2m/lwm2m_object.h:694]: nit: Please fix the alignment

File: subsys/net/lib/lwm2m/lwm2m_object.h [hunk 3]
```diff
@@ -736,6 +746,12 @@ static inline int engine_get_string(struct lwm2m_input_context *in,
 	return in->reader->get_string(in, buf, buflen);
 }
 
+static inline int engine_get_time(struct lwm2m_input_context *in,
+				   int64_t *value)
```
Comment (rlubos) [inline subsys/net/lib/lwm2m/lwm2m_object.h:750]: nit: Please fix the alignment

**Review by rlubos — COMMENTED** [review]
Looks good, just some nits.

**Review by rlubos — APPROVED** [review]

File: subsys/net/lib/lwm2m/lwm2m_object.h (hunk 1 above)
Comment (VeijoPesonen) [reply subsys/net/lib/lwm2m/lwm2m_object.h]: Assumably fixed.

File: subsys/net/lib/lwm2m/lwm2m_object.h (hunk 2 above)
Comment (VeijoPesonen) [reply subsys/net/lib/lwm2m/lwm2m_object.h]: Assumably fixed.

File: subsys/net/lib/lwm2m/lwm2m_object.h (hunk 3 above)
Comment (VeijoPesonen) [reply subsys/net/lib/lwm2m/lwm2m_object.h]: Assumably fixed.

**Comment by VeijoPesonen** [comment]
@SeppoTakalo Please review.

**Review by SeppoTakalo — APPROVED** [review]

**Pull request merged**